pkg/controller/flinkapplication/flink_state_machine.go

```diff
@@ -720,9 +860,25 @@ func (s *FlinkStateMachine) handleApplicationRunning(ctx context.Context, applic
 				fmt.Sprintf("Changing deployment mode from %s to %s is unsupported", application.Status.DeploymentMode, application.Spec.DeploymentMode))
 			return s.deployFailed(application)
 		}
-		logger.Infof(ctx, "Application resource has changed. Moving to Updating")
 		// TODO: handle single mode
-		s.updateApplicationPhase(application, v1beta1.FlinkApplicationUpdating)
+
+		// if our scale mode is InPlace and this is a suitable update (the only change is an increase in parallelism)
+		// move to Rescaling
+		if application.Spec.ScaleMode == v1beta1.ScaleModeInPlace && isScaleUp(application) {
+			if application.Spec.DeploymentMode == v1beta1.DeploymentModeBlueGreen {
+				s.flinkController.LogEvent(ctx, application, corev1.EventTypeWarning, "NotScalingInPlace",
+					"Not using configured InPlace scaling mode because it is incompatible with BlueGreen"+
+						" deploy mode")
+				s.updateApplicationPhase(application, v1beta1.FlinkApplicationUpdating)
+			} else {
```

Comment:
I think this `else` clause should be on one outer-level? Otherwise now, I see the `Application scale has increased. Moving to Rescaling.` even when scaling down.

Reply:
The first condition (on line 867) should only be true if this is a scaleUp, as implemented in `isScaleUp`, which checks that this is an increase in parallelism. If it's not a scale up, we should run the else block on line 878. The inner condition is just about checking for blue/green mode and giving a helpful error message in that case.

Reply:
You're right. My bad.